python/ambassador/config/config.py

```diff
@@ -345,25 +346,17 @@ def post_notice(self, msg: str, resource: Optional[Resource]=None, log_level=log
 
         self.logger.log(log_level, "%s: NOTICE: %s" % (rkey, msg))
 
-    @multi
-    def post_error(self, msg: Union[RichStatus, str], resource: Optional[Resource]=None, rkey: Optional[str]=None, log_level=logging.INFO) -> str:
-        del resource    # silence warnings
-        del rkey
-
-        if isinstance(msg, RichStatus):
-            return 'RichStatus'
-        elif isinstance(msg, str):
-            return 'string'
-        else:
-            return type(msg).__name__
+    @singledispatchmethod
+    def post_error(self, msg: Union[RichStatus, str], resource: Optional[Resource]=None, rkey: Optional[str]=None, log_level=logging.INFO) -> None:
+        assert False
```

Comment:
This is here as a fail safe to catch bugs correct? It should never happen if type checking stops user from calling post_error with anything other than RichStatus and str, correct? If it does then we know there is bug...